Storage/Storage/CoreData/CoreDataManager.swift

```diff
@@ -116,12 +107,12 @@ public final class CoreDataManager: StorageManagerType {
     public func performAndSave(_ operation: @escaping (StorageType) -> Void,
                                completion: (() -> Void)?,
                                on queue: DispatchQueue) {
-        let derivedStorage = writerDerivedStorage
+        let derivedStorage = writerStorage
         writerQueue.addOperation(AsyncBlockOperation { done in
             derivedStorage.perform {
                 operation(derivedStorage)
 
-                derivedStorage.saveIfNeeded()
+                (derivedStorage as! NSManagedObjectContext).saveIfNeeded()
```

Comment:
Intentionally using forced unwrapping here as we don't expect the writer storage to be of any other type in `CoreDataManager`.